src/libstd/net/udp.rs

```diff
@@ -295,6 +295,15 @@ impl UdpSocket {
     /// This will retrieve the stored error in the underlying socket, clearing
     /// the field in the process. This can be useful for checking errors between
     /// calls.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::net::UdpSocket;
+    ///
+    /// let socket = UdpSocket::bind("127.0.0.1:34254").expect("couldn't bind to address");
+    /// socket.take_error().expect("no error was expected");
```

Comment:
Can you just assert for either `Ok(None)` or `Ok(Some(something))` or `Err(something)`? This is very mind-bending.